@here/harp-mapview/lib/ClipPlanesEvaluator.ts

```diff
@@ -25,6 +25,10 @@ export interface ViewRanges {
      * @note Should be always positive and bigger then [[near]] plane distance.
      */
     far: number;
+    /**
+     * Minimum viewing distance that may apply to near plane.
```

Comment:
What does this mean?

Reply:
I have added some more notes to documentation. Quickly this is required setting for any rendering Api, cause we can not set near plane to zero, so usually it will hold some "epsilon" number, but we leave the flexibility to the user. 
The entire concept of ViewRanges is quite known in rendering engines cause it allows several effects to be implemented, for example imagine that we want to render real geometry in some range but put some special effects far behind or before, this is one of possible scenarios... He we use those ranges for fading and fog calculations that need to be tilt independent, this means that we can not base this effects on pure near/far planes, because those will be adjusted whenever tilt changes in order to preserve depth buffer precision.

Reply:
> so usually it will hold some "epsilon" number, but we leave the flexibility to the user.

Why does an epsilon value need to be configurable

If you want to use the maximum for fog, I think best rename it to something like `fogStart` or equivalent.

Reply:
> > so usually it will hold some "epsilon" number, but we leave the flexibility to the user.
> 
> Why does an epsilon value need to be configurable

Epsilon may be dependent from the depth buffer precision (and format, int32, float) and thus rendering Api, moreover when processing map features (geo-projection) we may still notice some inaccuracy and rounding errors, if you would like to see the effect change the margin to some very small value, suddenly you will note that some of the features may be missing, even though the far plane lays exactly on the ground level (best to test with TopView - AltitudeBased evaluator).  
> If you want to use the maximum for fog, I think best rename it to something like `fogStart` or equivalent.
Unfortunately we may not relate maximum visibility range to fog, because there are certainly more features that depends on that, take Labels fading or geometry fading as example. Maximum visibility is the distance that far plane may be placed on at extreme tilt angles, we will never see more then that, but for top-view far plane is just below the ground so it is not our visibility limit, but rather far plane limit for that tilt angle, if you rotate camera to horizon far plane moves further, but max visibility remains the same. On the old (legacy) evaluator max_visibility == far plane (ALWAYS), because we never thought about depth buffer precision and shrinking the camera frustum to minimum. Since now, we differentiate this two values, because we want to have frustum as small as possible at certain tilt angles, of course at critical angles (tilt) this values will equalize.